packages/common/utils/src/reverseKeyValue.ts

```diff
@@ -1,4 +1,14 @@
 /** @tossdocs-ignore */
+
 export function reverseKeyValue<KeyType extends string, ValueType extends string>(obj: Record<KeyType, ValueType>) {
-  return Object.fromEntries(Object.entries(obj).map(([key, value]) => [value, key])) as Record<ValueType, KeyType>;
+  const reversedObj: Record<ValueType, KeyType | KeyType[]> = {} as Record<ValueType, KeyType | KeyType[]>;
+  for (const [key, value] of Object.entries(obj) as Array<[KeyType, ValueType]>) {
+    if (value in reversedObj) {
+      const existingValue = reversedObj[value];
+      reversedObj[value] = Array.isArray(existingValue) ? [...existingValue, key] : [existingValue, key];
+    } else {
+      reversedObj[value] = key;
+    }
+  }
+  return reversedObj;
```

Comment:
I think this API`s purpose not change object format, but revert key with value.

Since the object with the same key value is overwritten at the end, I think it is safe to say that the current return value is not a bug but an intended action.

<img width="166" alt="image" src="https://github.com/toss/slash/assets/57122180/ab9139d7-79ac-4f6d-928e-3b905f0ba733">

So, I think this return value is not wanted by user.

Reply:
Thank you for your response. I sincerely appreciate it.
I consider this reply to be regarding the issue addressed in https://github.com/toss/slash/issues/235.
Unfortunately, I had to proceed with the task without knowledge of the initial intention and history of the method.
The fact that the issue from May is still open led me to believe that assistance was desired.
In the future, when making contributions to this open-source project, should I seek feedback on the desired type of PR in the corresponding issue before proceeding?
I am completely fine if this PR does not get accepted.
I believe that contributing to the well-structured monorepo of 'slash' would not only enhance my learning experience but also be beneficial to me.
If it is deemed unnecessary, would it be acceptable to close the PR?

Reply:
>Since the object with the same key value is overwritten at the end, I think it is safe to say that the current return value is not a bug but an intended action.

**I think it is good solution.**

If there is a duplicate key value, it seems to be the intention of object in `javascript`, seeing that it is overwritten with the last value used in order